### PR TITLE
deps: update x/tools and gopls to dd1c46858

### PR DESCRIPTION
**Pull request by leitzler**
Note that this change also contain required refactorings due to internal changes in gopls.

* gopls/internal/lsp/source: simplify extracting object hover doc dd1c46858
* gopls/internal/lsp/source: use syntax alone in FormatVarType 66f8f7199
* internal/imports: update stdlib index for Go 1.20 30f191f36
* internal/imports: use go/packages instead of cmd/api to compute symbols 4e98188bf
* internal/imports: update stdlib index for 1.20 4e8ff892a
* gopls/internal/lsp: go to definition from linkname directive 6bd0d0048
* gopls/internal/lsp: enable clear builtin completion test 0cfddb3c3
* gopls/internal/lsp/tests: remove StripSubscripts 41adf8d4f
* gopls/internal/lsp/safetoken: delete safetoken.Range 86fdadc60
* internal/robustio: fix signature of getFileID on plan9 c276ee52b
* gopls/internal/lsp: add clear builtin e170d456f
* go/ast/inspector: skip ranges that do not contain a node type 2ea4b81bf
* go/analysis: improve error message on duplicate fixes 407bbed8c
* all: fix some comments bd5dfbb41
* gopls/protocol: use the current definition of the lsp 072fca58f
* tools/gopls: provide markdown for completion and signature help aa633e7ed
* go/analysis/internal/analysisflags: use os.Executable for program path 684a1c0d5
* gopls/internal/lsp/cache: add missing mutex bd5e595be
* gopls/internal/lsp: differentiate govulncheck/vulncheck imports diags 26831282a
* gopls/internal/lsp/mod: reorder vulncheck quick fixes d1e92d6ab
* gopls/internal/lsp: separate some requests from source.Identifier 87d00e630
* gopls: fix windows file corruption ae242ec32
* gopls/internal/lsp/protocol: Mapper.NodeMappedRange 6f652132c
* gopls/semantic: report type parameters in the type of a receiver e26036888
* internal/lockedfile: fix build constraints on solaris b62cbb6b5
* gopls/internal/lsp/source: avoid qualifiedObjectsAtProtocolPos 1aa7e72e7
* gopls/internal/lsp/source: rename: prep for incrementality 5ed33dffa
* gopls/internal/lsp/source: simplify legacy 'references' func e0659d1f3
* gopls/internal/regtest/diagnostics: require cgo for TestGoListErrors 1edcfe78f
* gopls/internal/lsp/protocol: move TestBytesOffset f05215879
* gopls/internal/lsp/protocol: LocationTextDocumentPositionParams d093a1395
* gopls/internal/regtest: make RegexpSearch return a Location bcb677e49
* gopls/internal/lsp/source: eliminate a couple uses of posToMappedRange 60782e9bd
* gopls/internal/lsp/source: eliminate ResolveImportPath 031e6e6d0
* gopls/internal/lsp/source: reduce usage of TypecheckWorkspace f2cd9ef6a
* gopls/internal/lsp/cache: remove package dependence on packages.Config f10e7d56a
* gopls/internal/lsp/source: switch call hierarchy to references v2 827075753
* gopls/internal/lsp/source: references: support exported methods 37c69d87c
* gopls/internal/lsp/cmd/test: delete marker-based tests of gopls cmd f3c36a209
* gopls/internal/lsp/cmd/test: new integration test for gopls command c224aaef0
* gopls/internal/lsp/source/xrefs: allow Lookup of a set deeb64bdf
* gopls/internal/lsp: remove Server.processModifications f269f537b
* gopls: allow 'any' and 'comparable' in completion results fcd57ebb7
* gopls/internal/lsp/source: referencesV2: support unexported methods aae3642ea
* tools/gopls: register semantic tokens statically b5d65e025
* gopls/internal/lsp/source: stub: don't panic when encountering 'error' 51abc5bd8
* gopls/internal/regtest: add a test demonstrating confusion following an options change ce28f407b
* internal/lsp/cache: don't panic in Snapshot on a shutdown view c4c6aa640
* tools/internal/diff: fix off-by-one error in computing diffs 1faecd32c
* gopls/internal/lsp: consolidate the FileHandle API a7f033afe
* internal/lockedfile/internal/filelock: fix aix build tag 271e621c9
* gopls/internal/lsp/cmd/test: signpost future cleanups ff9bea528
* gopls/release: remove unused functionality from release script 7d4ba2fe8
* gopls/internal/lsp/source: delete source_test 46b69584a
* go/analysis/passes/directive: add directive analyzer bcc7794c0
* gopls/internal/lsp: add missing comments on 3x tests.Test impls 33d416e14
* gopls/internal/lsp/source: make implementations2 work on embedded fields afea27258
* gopls/internal/lsp/filecache: use file locking, not rename bb7c440ef
* gopls/internal/lsp/filecache: actually delete files 561a9be67
* gopls/internal/lsp/source: delete IsInterface 9682b0d47
* go/analysis/passes/loopclosure: avoid panic in new parallel subtest check when t.Run has single argument 7a7b699cd
* gopls/internal/regtest: use AfterChange in more places 3e6f71bba
* x/tools/go/analysis/passes/printf: revert URL in error message 9ff31a578
* gopls/internal/lsp/source/impls: a new "implementations" index 2fa6ca1e2
* gopls/protocol: new versions of generated LSP files 957bec5ae
* internal/gcimporter: discard position info for unneeded lines f0e2d5c52
* cmd/digraph: use ReadString rather than bufio.Scanner 5bedd861b
* gopls/internal/lsp: remove the experimentalWatchedFileDelay setting f6ea009c5
* gopls/internal/lsp/source: include ITVs in global references f46e4185a
* internal/jsonrpc2_v2: fix typos f3e53e5cc
* internal/gcimporter: use two-level file index d958e8548
* internal/gcimporter: preserve column and line in shallow iexport 8aba49bb5
* gopls: new LSP stub generator d7fc4e711
* internal/robustio: skip os.Link test on android 5c176b1de
* go/ssa: sanity check the types of phi nodes d34a055ab
* go/callgraph/vta: add flows for receiver function types 6f095b4f2
* cmd/fiximports: do not assume go list -json unmarshals into build.Package 8e9496736
* go/ssa: fix phi node type in slice to array conversion e035d0c42
* go/expect: remove testdata go.mod to go.fake.mod 03eac8179
* gopls/internal/regtest: follow-ups to review comments from earlier CLs 1e819a3cf
* gopls/internal/regtest: clean up workspace symbol helpers 9ba8bb1a1
* gopls/internal/regtest: eliminate DiagnosticAtRegexp 91b60705d
* gopls/internal/regtest: eliminate DiagnosticsAtRegexpWithMessage bd48b9a5d
* gopls/internal/regtest: eliminate DiagnosticAt 5d65394a9
* gopls/internal/regtest: replace NoDiagnostics with NoMatchingDiagnostics 27dfeb267
* gopls/internal/lsp/fake: use protocol types for the fake editor 87092c8c8
* gopls/internal/regtest: simplify OnceMet expressions with an env helper 672a036fc
* gopls/internal/regtest: eliminate GoSumDiagnostic ab7b5b24f
* gopls/internal/regtest: add a simpler API for diagnostic expectations 331a1c689
* gopls/internal/regtest: eliminate EmptyDiagnostics c9b82f2d1
* gopls: update golang.org/x/vuln@6ad3e3d e81af2785
* internal/regtest/bench: fix BenchmarkRename and add more benchmark tests for gopls d19e3d18a
* gopls/internal/regtest: rename EmptyOrNoDiagnostics to NoDiagnostics 2be1a9a65
* gopls/internal/regtest: eliminate NoDiagnostics 7ec05ac77
* gopls/internal/regtest: eliminate DiagnosticsFor e956495a7
* gopls: remove the experimentalWorkspaceModule mode 808791186
* gopls/internal/lsp/cache: clean up view workspace information 5b300bd64
* gopls/internal/cache: don't mark initialized after cancellation 97d5de5ad
* gopls/internal/lsp/glob: add an LSP compliant glob implementation 58691bc25
* cmd/cover: delete package a3c22fca0
* cmd/cover: remove replace directive 98dcb0ee0
* gopls/internal/lsp/source: minor clarifications 77655679d
* cmd/cover: carve out deprecated command into its own module a7f7db3f1
* Revert "cmd/cover: carve out deprecated command into its own module" f9a10c01c
* internal/gcimporter: fix export of invalid methods e345d46ec
* gopls/internal/lsp/cache: don't cache files if mtime is too recent 4305a2291
* internal/regtest/misc: fail eagerly in TestRenameFileFromEditor 227ee72c9
* cmd/cover: carve out deprecated command into its own module 43158af4a
* gopls/internal/lsp/protocol: cleanups and docs for Mapper b79893424
* gopls/internal/lsp/protocol: rename s/ColumnMapper/Mapper/ a24944e25
* gopls/internal/span: simplify Span 55935f417
* gopls/internal/lsp/lsppos: delete Mapper 40a1c97a3
* gopls/internal/lsp/protocol: reimplement ColumnMapper line logic 6a3bc37a9
* go/callgraph/cha: refactor callee construction 6e9a35db6
* go/callgraph: fix slicing in callgraph_test.go fef5b76e4
* gopls/internal/lsp/source/xrefs: a new reference index 2be9d05fe
* gopls/internal/lsp/lsppos: delete TokenMapper 0362ceaa7
* go/callgraph/vta: optimize scc computation 67baca651
* gopls/internal/lsp/filecache: use TempDir if UserCacheDir fails us 2eb613824
* gopls/internal/lsp/protocol: move MappedRange 36bd3dbcc
* gopls/internal/lsp/cache: assume Go 1.16+ 16b3bf8d4
* internal/robustio: add Plan9 support to FileID 3856a5d80
* gopls/internal/lsp/fake: fix EOF bug in applyEdits 09cbc4219
* gopls/internal/lsp/cache: use distinct types for mod and work parse keys 4ded35d13
* gopls/completion: avoid duplicating text in test func completions 107f43f66
* gopls/internal/lsp/mod: fix nil panic in go.mod hover e225fd43c
* gopls/internal/lsp/filecache: use os.Chtimes 057ed3c5c
* gopls/internal/lsp/source: MappedRange cleanup 1fe76afbb
* gopls/internal/lsp/protocol: simplify ColumnMapper 02bea03ba
* go/callgraph: adds benchmarks comparing algorithms a4455febd
* go.mod: update golang.org/x dependencies 7db99dd12
* gopls/internal/regtest: avoid race in TestSwitchFromGOPATHToModuleMode 1e0dff28f
* gopls/internal/lsp/cache: use specific mutexes for module data 0441b432c
* internal/robustio: move robustio 33071fbe1
* gopls/internal/regtest/watch: don't run TestSwitchFromGOPATHToModuleMode in experimental mode b01e7a4e7
* gopls: remove dead analysis code e417ea36b
* gopls/internal/lsp: update replace directives in go.mod for package renaming 1a08d01ac
* gopls/internal/regtest: port experimental workspace tests to go.work eac36cb2a
* gopls/internal/lsp/source: delete Snapshot.WriteEnv method 224a61b35
* gopls/internal/lsp/safetoken: funnel more calls through this package 81e741e32
* gopls/internal/regtest: await go.work changes in TestAddAndRemoveGoWork 8367fb272
* gopls/internal/regtest: make BufferText strict 3b16059a0
* gopls/internal/lsp/cache: recreate Views when their root changes 0e1d01321
* go/ssa,go/analysis/passes/nilness: refine when type param constants are nil 2f31dd473
* gopls/internal/lsp/source: delete GetTypedFile ae4ff823b
* gopls/internal/lsp/source: eliminate Snapshot.Package{,s}ForFile fe6b3007e
* gopls/internal/lsp/cache: eliminate snapshot.containingPackages 26fc60977
* gopls/internal/lsp/safetoken: fix bug in Offset at EOF 85e6ad773
* gopls/internal/lsp/safetoken: fix error message ef1ec5d78
* gopls/internal/lsp/source: avoid unnecessary transitive rdeps 44395ffdd
* Revert "gopls/internal/regtest: harmless CL used for benchmark test" 6546d82b2
* gopls/symbols: call source.Document symbols only for Go files 3be06475e
* gopls/internal/lsp: Replace input text when completing a definition d462c83f2
* gopls/internal/regtest: harmless CL used for benchmark test 7efffe157
* gopls/internal/lsp: more comment tweaks post-//line support 1627e959e
* internal/lsp/debug: fix broken template 21f61007c
* gopls/internal/robustio: FileID, a portable file identifier 6ad27d09a
* internal/diff/lcs: optimize inner loop 6df6eee46
* go/gcexportdata: drop support for go1.6 export data 57b126584
* gopls/internal/lsp: followups to dropping line directives 099260e6e
* gopls/internal/lsp/cache: a new analysis driver 61e2d3f4f
* gopls: ignore //line directives eb70795aa
* go/ssa: deref core type in emitLoad b4dfc3609
* gopls/internal/lsp: announce selectionRangeProvider capability 1270fd75a
* gopls/internal/lsp/cache: simplify DiagnosePackage 9bc5dce4d
* x/tools: drop support for Go toolchains older than go1.16 df35cd84e
* go/gcexportdata: preallocate buffer to read into when size is known db9d10fc1
* gopls/internal/lsp: document analysis-related functions 0d2045bac
* gopls/internal/lsp/cache: reduce type-checking in renameImports b2b9dc3de
* go/ssa/interp: fix conversion of slice to named array 3cb82d5f9
* gopls: upgrade dependencies following release 5899b6a2b
* gopls/internal/lsp/cache: delete Snapshot.KnownPackages 763a03075
* gopls/internal/hooks: panic if diff consistency check fails cc0e696e5
* gopls/internal/lsp/source: emit interface stub methods at top level 9ec855317
* gopls/internal/lsp/cache: only invalidate parsed files if they changed 444c8f69a
* gopls/internal/lsp/source: use correct token.File 601ca6ce8
* internal/lsp/mod: fix run_govulncheck codelens name 95c9dce0e
* gopls/internal/lsp: add selection range request d72a64a96
* gopls/internal/lsp: split ActivePackages 18f76ecd3
* gopls/internal/lsp/cache: simplify ad-hoc package warning logic 84299a05d
* gopls/internal/lsp/cache: record parse keys when they're created a3eef2595
* gopls/hover: remove header tags from hover markdown 3da7f1e4c
* gopls/internal/lsp/source: eliminate getStubFile a310bcb1d
* internal/gcimporter: port CL 424876 from std importer 3cba5a847
* gopls/internal/lsp/mod: add Reset vulncheck result codelens b0fdb78b2
* gopls/internal/lsp: perform analysis unconditionally 88ceb2401
* gopls/internal/lsp/cache: invalidate govulncheck results older than 1hr 3f74d914a
* gopls/internal/lsp/source: avoid typechecking in findRune 6b8674f10
* gopls/internal/lsp: eliminate more unnecessary typechecking d7dfffd6c
* gopls/internal/lsp/fake: use robustio.RemoveAll in (*Workdir).RemoveFile f3fb218e7
* gopls/internal/vulncheck: add TODO for the vulncheck diagnostics 96ff41d58
* gopls: delete obsolete govulncheck code and stop gopls vulncheck -summary 0f6c6f11e
* gopls/internal/lsp/regtest: fix TestRunGovulncheckError2 c5343a6ac
* gopls/internal/lsp: avoid type-checking when computing hyperlinks cb701f7c7
* gopls/internal/lsp/source: avoid unnecessary calls to GetTypedFile d0f184db7
* gopls/internal/lsp/fake: simply use polling to simulate file watching 1e5efed1b
* gopls/internal/lsp/fake: eliminate the unnecessary fake.FileEvent 838a165e4
* gopls/internal/lsp/fake: eliminate the unnecessary ChangeFilesOnDisk API 09fb680dd
* gopls/internal/lsp/source: KnownPackagePaths: avoid loading 09ae2d5af
* gopls/internal/lsp/cache: split metadata and loading in PackagesForFile 1dcc423c1
* gopls/release: fix the release script when go.work is not used 6b505011c
* gopls/internal/lsp/fake: in (*Workdir).RenameFile, fall back to read + write aee3994bd
* go.mod: update golang.org/x dependencies fe60148df
* gopls/internal/regtest: add a test for the case when the renaming package's path contains "internal" as a segment c9ea9a72c
* gopls/internal/lsp/cache: improve ad-hoc warning for nested modules bf5db8100
* go/analysis: document that facts are gob encoded in one gulp aa9f4b2f3
* internal/gcimporter: skip tests earlier when 'go build' is not available bdcd08225
* gopls/internal/lsp/cache: expand ImportPath!=PackagePath comment 2ad6325d9
* gopls/internal/robustio: only define ERROR_SHARING_VIOLATION on Windows 52c7b88fe
* gopls/internal/lsp/cache: narrow reloadOrphanedFiles to open files 4f69bf3eb
* gopls/internal/regtest/misc: test Implementations + vendor 6002d6ea5
* internal/gcimporter: load cached export data for packages individually f540ee6b1
* gopls/internal/lsp/cache: simplify canonical URI cache d444fa3b3
* gopls/internal/regtest/misc: skip vendor test on go1.13 25fdb81eb
* gopls/internal/lsp/cache: invalidate metadata after vendor change e0b516bc0
* gopls/internal/lsp/cache: fix re-entrant session locking 31d584333
* gopls/internal/vulncheck: always use pkg.go.dev/vuln for urls 8c78b306b
* gopls/internal/regtest/misc: skip TestRunGovulncheckError2 47a82463d
* gopls/internal/lsp: avoid I/O in URI comparison operations d54e12bfc
* internal/gcimporter: fix TestImportStdLib 0379b73ae
* gopls/internal/vulncheck: handle package errors e79e423fe
* gopls/internal/lsp/mod: disable the diagnostics on stdlib vulns c5ce80693
* gopls/internal/vulncheck: clarify the log message 78c1861cf
* gopls: move reset go.mod diagnostic codelens to module statement 1a0053c95
* gopls/internal/regtest: fix TestRunGovulncheckStd 9b8d87b59
* gopls/internal/lsp: add run vulncheck fix for stdlib vulns 81b6bef4d
* gopls/internal/lsp: move options off of the cache fe83ddb67
* gopls/internal/lsp/source: enable run_govulncheck codelens in exp mode 88ee30bcc
* gopls/internal/lsp/command: rename run_vulncheck_exp to run_govulncheck 0a6aa90a5
* gopls/internal/lsp/mod: add hover for stdlib vulnerabilities bedad5a8e
* gopls/internal/vulncheck: check vulnerabilities in standard library 7a1b5707a
* gopls/internal/lsp/mod: add "Run govulncheck" codeaction 9a54670c1
* gopls/internal/lsp: add an option to overwrite previous diagnostics f1b76aef9
* gopls/internal/lsp: remove access to mutable session state from the view e8a70a5e3
* gopls/internal/lsp/cache: eliminate obsolete invalidation step ff22fab93
* gopls: add a go:generate rule for doc generation 3881046b0
* internal/gocommand: add GoVersionString 4c3cb1e65
* gopls: add diagnostic for standard library e29d1ed00
* gopls/internal/lsp: include all vulns info to fetch_vulncheck_result f718365b8
* gopls/internal/lsp/mod: add the vulncheck diagnostics mode 9519368a4
* gopls/internal/govulncheck: add in-memory cache for vulndb 7d9f45139
* go/ssa: deprecate coreType and _NormalTerms 1ccccf8da
* gopls: add diagnostics for both vulns used and not used 4b7d1b311
* internal/regtest/bench: add benchmark tests for gopls 50ccef5d0
* gopls/internal/lsp/source: Package clarifications d39685aa6
* gopls/internal/vulncheck: add import info based vuln scanning 7cfde84b7
* internal/gcimporter: adjust the number of expected packages in TestStdlib 2b56641c3
* gopls: skip vulnerabilities that are probably fixed in hover 50df761c6
* gopls/internal/lsp/cache: wire in mod vulnerability analysis 7cda55e58
* gopls: add quick fixes for all vulnerabilities 5c3561767
* gopls/internal/lsp/command: return the vulncheck progress token fb7be641d
* go/packages: fix doc typo 060c049c4
* gopls: fix formatting for vulncheck results in hover 6eafd96cb
* go/analysis/passes/loopclosure: recursively check last statements in statements like if, switch, and for b797704ca
* internal/gcimporter: in TestStdlib, only check x/tools packages if we expect to have their source 3b9d20c52
* go/packages: change workaround for issue 28749 to not require NeedFiles 2ad3c3337
* gopls/internal/lsp/source: avoid panic(nil) 57f56abcb
* internal/lsp/source: avoid using go/types.Implements with Typ[Invalid] 41c70c91b
* analysis: correct go/analysis/passes/findcall path in docs db5eae217
* cmd/godoc: streamline test subprocesses b978661c6
* internal/testenv: add a Command function that replaces exec.Command 932ec22a5
* gopls/internal/lsp/cache: fix data race in Symbols 19fb30d1d
* gopls/internal/lsp: add gopls.fetch_vulncheck_result 4ce4f93a9
* gopls/internal/lsp/mod: highlight when there is no fix for vuln 8ba9a370e
* gopls/internal/lsp/mod: add related info 128f61d43
* internal/lsp/mod: adjust vulncheck diagnostics suppression logic fc039936a
* gopls/internal/vulncheck: log progress c099dff17
* go/ssa: build generic function bodies 36a5c6a8a
* gopls/internal/lsp/source: eliminate Metadata interface 85bf7a8fb
* gopls/internal/lsp: simplify KnownPackages 2592a854e
* gopls/internal/lsp/cache: clean up tracking of GO111MODULE c7ed4b3c0
* internal/lsp/cache: clean up getOrLoadIDsForURI 23056f613
* internal/lsp/cache: remove support for invalid metadata 5a4eba5af
* gopls/internal/lsp/mod: fix unaffecting vuln diagnostics msgs 32a17c01d
* internal/testenv: skip tests that need export data for std if 'go tool compile' does not exist dea100b11
* playground/socket: eliminate an arbitrary timeout in TestLimiter ba373eed3
* gopls/internal/lsp/lsprpc: eliminate arbitrary timeout in TestEnvForwarding 3d085f3fc
* gopls/internal/lsp/regtest: improve documentation 434d569df
* go/analysis: add Pass.TypeErrors field ce26db420
* gopls/internal/regtest/misc: fix use of the AfterChange API b0ad6b2e0
* gopls/semantic: semantic tokens for type parameters 89856a4c9
* internal/pkgbits: port small optimization from compiler to tools 6e8da3feb
* internal/gcimporter: port memory reuse optimizations from Go tree 06fb723d2
* internal/gcimporter: fix performance regression for unified IR fc702c522
* gopls/internal/regtest: make AfterChange do the awaiting 1cb4c1798
* gopls/internal/lsp: fix diagnostic suppression when folders change 0c71b564b%

### CODE REVIEW

**Comment by myitcv** [comment]
Thanks for working on this, @leitzler !

File: cmd/govim/internal/types/point.go [hunk 1]
```diff
@@ -26,66 +26,57 @@ type Point struct {
 }
 
 func PointFromOffset(b *Buffer, offset int) (Point, error) {
-	tf := b.tokFile()
-	line, col, err := span.ToPosition(tf, offset)
+	m := protocol.NewMapper(b.URI(), b.contents)
```
Comment (myitcv) [inline cmd/govim/internal/types/point.go:29]: My only concern here (and below) would be that we create a new mapper every time we call this function. Previously, there was lazy creation and then reuse, no?

**Review by myitcv — COMMENTED** [review]
Generally LGTM with one comment about allocation/reuse of a mapper

File: cmd/govim/internal/types/point.go (hunk 1 above)
Comment (leitzler) [reply cmd/govim/internal/types/point.go]: You are absolutely right, I fixed it.

**Comment by leitzler** [comment]
PTAL, in case you missed my re-request :)

**Review by myitcv — APPROVED** [review]
LGTM

**Pull request merged**